### ISSUE TITLE
Variables are shown in random order

### ISSUE THREAD

**Issue by jukzi** [eclipse-oomph/oomph#16]
every once i start Oomph the Variables are shown in a different order.
Example1:
![image](https://github.com/eclipse-oomph/oomph/assets/51790620/71ba7bfd-45f8-4308-a9ab-fb096127a5c8)
Example2:
![image](https://github.com/eclipse-oomph/oomph/assets/51790620/6b91290c-1b9b-4423-b101-0faa76c354c3)

Can even be triggered by using  "back" and "next" buttons.
That's irritating.

**Comment by merks** [comment]
The only way I've been able to get the second order is to check and uncheck "Show all variables".  

It looks to me like the logic for the order is deterministic.  There is some reordering involved though to keep "related" variables closer together and to keep impactful variables near the beginning.  You're really sure this really comes up randomly differently initially or that it randomly changes for next/back navigation?    I really cannot reproduce that, and not for lack of trying...

Note too that I assume you get to this with the platform configuration, but after a dozen clicks on that all with the same "picture number one", I'm not convinced there is any randomness.  Please double check your steps to reproduce the issue.

**Comment by merks** [comment]
If I cannot reproduce it then I cannot fix it.

**Comment by mmosconi1** [comment]
The order of variables in the dialog seems to depend on the order of their usage in the setup.
That's why we try to get a defined order by introducing an internal variable referencing all user variables in the desired order and put this one first in the setup.
But even with this approach, the variables that are used in filters (bold label) are automatically placed. The same for variables inside conditional containers.

**Comment by merks** [comment]
Yes, the ones used in filters are intentionally moved to the front because they affect which tasks are collected and that in turn can affect which other variables are used and hence prompted/displayed...

If someone can provide something  or some steps where the order changes or appears different "each time" I can investigate.  But if I can't reproduce anything like it, I'm at a loss to make progress on it.  Sorry.  😱
